tools/c7n_azure/c7n_azure/resources/key_vault_keys.py

```diff
@@ -187,6 +191,76 @@ def process(self, resources, event=None):
             client = self.manager.get_client(vault_url=id.vault_url)
             rotation = client.get_key_rotation_policy(id.name)
             if (self.data.get('state') == 'Disabled' and not rotation.id) or \
-               (self.data.get('state')  == 'Enabled' and rotation.id):
+               (self.data.get('state') == 'Enabled' and rotation.id):
                 matched.append(key)
         return matched
+
+
+@KeyVaultKeys.action_registry.register("update")
+class KeyVaultKeyUpdateAction(AzureBaseAction):
+    """
+    Update a key's properties. This does not update its cryptographic material.
+
+    expires_on and not_before require isoformatted dates
+
+    :example:
+
+    .. code-block: yaml
+        policies:
+            - name: update-expires-on
+              resource: azure.keyvault-keys
+              filters:
+                - name: foo
+              actions:
+                - type: update
+                  expires_on: "2025-12-12 00:00:00"
+    """
+
+    schema = type_schema(
+        "update",
+        key_operations={
+            "type": "array",
+            "items": {
+                "type": "string",
+                "enum": [
+                    "decrypt",
+                    "encrypt",
+                    "export",
+                    "import_key",
+                    "sign",
+                    "unwrap_key",
+                    "verify",
+                    "wrap_key"
+                ]
+            },
+        },
+        enabled={"type": "boolean"},
+        tags={"type": "object"},
+        not_before={"type": "string", "format": "date-time"},
+        expires_on={"type": "string", "format": "date-time"},
+    )
+
+    def _process_resource(self, resource):
+        name = resource['vault_id']['resource_id']['name']
+
+        id = KeyProperties(key_id=resource['id'])
+
+        client = self.manager.get_client(vault_url=id.vault_url)
+
+        not_before = self.data.get("not_before")
+        expires_on = self.data.get("expires_on")
+
+        if not_before:
+            not_before = datetime.datetime.fromisoformat(not_before)
+
+        if expires_on:
+            expires_on = datetime.datetime.fromisoformat(expires_on)
```

Comment:
It looks like using the `date-time` jsonschema type for these keys accepts any string whether or not it's a valid date :thinking: . That may be related to the validator we're using. In local testing I can fail validation if I put a number/list/etc, but any string seems to pass. In any case, I'm thinking any validating/parsing should probably move earlier (perhaps to `validate()`) to catch issues early and avoid reparsing the same value for each resource.

Using dateutil.parser or [parse_date](https://github.com/cloud-custodian/cloud-custodian/blob/3618740fd61942e7b69cff72056ebd7fe9c9d65d/c7n/utils.py#L120) from utils may avoid some surprises here too.